src/Build/Graph/ProjectGraph.cs

```diff
@@ -85,6 +86,7 @@ public ProjectGraph(string entryProjectFile)
                 globalPropertyDictionary);
             EntryProjectNode = _allParsedProjects[new ConfigurationMetadata(entryProjectFile, globalPropertyDictionary)];
             ProjectNodes = _allParsedProjects.Values;
+            DetectCyclesInGraph();
```

Comment:
>DetectCyclesInGraph(); [](start = 12, length = 22)

This does another traversal of the graph after LoadGraph. I am working on combining the 2 so that cycle detection can be done in the same traversal in which projects are discovered